### PR TITLE
Swap emulator Node runtime discovery to favor local cache

### PR DESCRIPTION
**Pull request by abeisgoat**
Right now we always assume the healthiest node install you have is the global one, this isn't the case when running in firepit, so I've swapped the preference to prefer any version of node installed to `node_modules` (if it exists and matches the requested runtime version).

### CODE REVIEW

**Comment by abeisgoat** [comment]
This merge looks super ugly, I literally just moved one `if` above the other

File: src/emulator/functionsEmulator.ts [hunk 1]
```diff
@@ -672,6 +672,18 @@ export class FunctionsEmulator implements EmulatorInstance {
       disabled_features: this.args.disabledRuntimeFeatures,
     };
   }
+  /**
+   * Returns a node major version ("10", "8") or null
+   * @param frb the current Functions Runtime Bundle
+   */
+  getRequestedNodeRuntimeVersion(frb: FunctionsRuntimeBundle): string | null {
```
Comment (samtstern) [inline src/emulator/functionsEmulator.ts:679]: Is there a reason this has to be `string | null` not `string | undefined`?  If you return `undefined` in the case of nothing then it's just:
```js
return frb.nodeMajorVersion || (pkg.engines && pkg.engines.node)
```

If you do need `null` then:
```js
const pkg = require(path.join(frb.cwd, "package.json"));
return frb.nodeMajorVersion || (pkg.engines && pkg.engines.node) || null
```

But this is all optional, method is clearly correct as-is just maybe overly complex?

**Review by samtstern — APPROVED** [review]
Small nits but LGTM

**Comment by samtstern** [comment]
@abeisgoat don't forget a `CHANGELOG` entry

**Pull request merged**